# HG changeset patch
# User Wolfgang Rosenauer
# Date 1558442915 -7200
#      Tue May 21 14:48:35 2019 +0200
# Node ID b467ec7989fed1c0639a59e90f7789234e3ed0a5
# Parent  9febcef65911f649028c28226d3beb194569375f
Do not use gconf for proxy settings if not running within Gnome
Index: toolkit/system/unixproxy/nsUnixSystemProxySettings.cpp
===================================================================
RCS file: /cvsroot/mozilla/toolkit/system/unixproxy/nsUnixSystemProxySettings.cpp,v
retrieving revision 1.1

diff -r 9febcef65911 -r b467ec7989fe toolkit/system/unixproxy/nsUnixSystemProxySettings.cpp
--- a/toolkit/system/unixproxy/nsUnixSystemProxySettings.cpp	Wed Jun 05 06:00:24 2019 +0000
+++ b/toolkit/system/unixproxy/nsUnixSystemProxySettings.cpp	Tue May 21 14:48:35 2019 +0200
@@ -55,11 +55,14 @@
 }
 
 void nsUnixSystemProxySettings::Init() {
-  mGSettings = do_GetService(NS_GSETTINGSSERVICE_CONTRACTID);
-  if (mGSettings) {
-    mGSettings->GetCollectionForSchema(
-        NS_LITERAL_CSTRING("org.gnome.system.proxy"),
-        getter_AddRefs(mProxySettings));
+  const char* sessionType = PR_GetEnv("DESKTOP_SESSION");
+  if (sessionType && !strcmp(sessionType, "gnome")) {
+    mGSettings = do_GetService(NS_GSETTINGSSERVICE_CONTRACTID);
+    if (mGSettings) {
+      mGSettings->GetCollectionForSchema(
+          NS_LITERAL_CSTRING("org.gnome.system.proxy"),
+          getter_AddRefs(mProxySettings));
+    }
   }
 }
 
